CMakeLists.txt

```diff
@@ -113,7 +112,7 @@ endif()
 add_library(${PROJECT_NAME} STATIC ${LIBCORO_SOURCE_FILES})
 set_target_properties(${PROJECT_NAME} PROPERTIES LINKER_LANGUAGE CXX PREFIX "")
 target_compile_features(${PROJECT_NAME} PUBLIC cxx_std_20)
-target_include_directories(${PROJECT_NAME} PUBLIC include)
+target_include_directories(${PROJECT_NAME} PUBLIC include vendor/tartanllama/expected/include)
```

Comment:
I'm curious why this was needed? Is this an MSVC problem? It doesn't feel like the right approach, especially since I'm working on a branch to use `std::expected` if the user has a C++23 compiler with support.

Reply:
Several files (`semaphore.hpp`, `ring_buffer.hpp`) have the statement `#include <tl/expected.hpp>`. I don't understand how this builds without this change.

Another option is to use the C++23 std::expected, but I still get errors on MSVC when I change this to `#include <expected>` and change `tl::expected` to plain `expected` in `ring_buffer.hpp`:

```
ring_buffer.hpp(149): error C7568: argument list missing after assumed function template 'expected'
ring_buffer.hpp(161): note: see reference to class template instantiation 'coro::ring_buffer<element,num_elements>::consume_operation' being compiled
```

Reply:
However, I'd be quite happy to use the C++23 approach in your branch if that works.

Reply:
I'd be happy to revert this line if you'd like; I think the rest of the PR is OK. We can come back to this.

Reply:
I think it is unnecessary.  If `LIBCORO_EXTERNAL_DEPENDENCIES=OFF` then the root CMakeLists.txt  around line 37 to 54 includes `add_subdirectory(vendor/tartanllama/expected)` which will set the include directory appropriately for `#include <tl/expected>`.  That feature was added a while ago so users could provide their own if they wanted to avoid git submodules or already had them in their project so they wouldn't have to be doubly included.  The user needs to set include paths correctly in _their_ build system though if they do this.

-----

The PR I'm working on for `std::expected` is here: https://github.com/jbaldwin/libcoro/pull/174 you can see how I made a compile time interface to pick one or the other... but the compilers don't seem to be setting `__cplusplus` versions properly so the build is failing :(

Reply:
> I think it is unnecessary. If `LIBCORO_EXTERNAL_DEPENDENCIES=OFF` then the root CMakeLists.txt around line 37 to 54 includes `add_subdirectory(vendor/tartanllama/expected)` which will set the include directory appropriately for `#include <tl/expected>`. `LIBCORO_EXTERNAL_DEPENDENCIES` feature was added a while ago so users could provide their own if they wanted to avoid git submodules or already had them in their project so they wouldn't have to be doubly included. The user needs to set include paths correctly in _their_ build system though if they do this.

OK, thanks -- I missed that when playing with the library. I'll update my PR to remove this change.